sphire/protocols/protocol_cryolo_napari_tomo_picking.py

```diff
@@ -65,29 +65,40 @@ def prepareDataStep(self):
         convert.convertMicrographs(tomoList, self._getExtraPath())
 
     def runCoordinatePickingStep(self):
-        # Getting the first tomogram to check if the .cbox file exist
-        tomogram = self.getInputTomos().getFirstItem()
-        filePath = os.path.join(self._getExtraPath(),
-                                convert.getMicFn(tomogram, "cbox"))
-        creationOldTime = None
-        if os.path.exists(filePath):
-            creationOldTime = time.ctime(os.path.getctime(filePath))
+        """Run napari-boxmanager"""
+
+        fileDict = {}
+        # Finding the coordinate file per tomogram
+        for tomogram in self.getInputTomos().iterItems():
```

Comment:
can we not use this very expensive function, especially multiple times? You only need getFileName from the tomo, which can be obtained with aggregate functions.